#2871: Adding Bar on top of Landing Page Indicating that User is on Test Server

MohitSoni11 opened the pull request:
@misaugstad My computer was doing weird things so I created a new Pull Request and closed the previous one.

Resolves #2871 

1. To resolve this issue, I added a bar on the top of the Landing Page notifying the user that they are on a test server rather than a production server.
2. If the user wants to stay on the test server, there is a close button that the user can press to get rid of the banner -- note, however, that the banner will reappear if the user reloads the page.
3. I have adjusted the banner formatting for smaller screen widths.

Banner Screenshot: 
![image](https://user-images.githubusercontent.com/92624795/172784698-743876e4-9804-4b67-a6a4-89fb4f85d683.png)

Code Explanation (Revised Code):
![image](https://user-images.githubusercontent.com/92624795/172784916-8d4d8cdd-0500-4a23-9523-6b41412af224.png)

- The if statements make sure that the banner is visible only when the user is on a test server
- The JavaScript at the bottom programs what happens when the button is pressed: the banner is removed.

misaugstad commented on app/views/navbar.scala.html:
```diff
@@ -11,9 +11,30 @@
     <div id="header-container" class="navbar-container container-fluid">
+        <!-- Banner is only visible if user is on test server-->
+        @if(Play.configuration.getString("environment-type").get == "test") {
+            <link href = "https://fonts.googleapis.com/icon?family=Material+Icons" rel = "stylesheet">
```
Can you edit the HTML to remove the space around the `=`? I've just added this to the [style guide](https://github.com/ProjectSidewalk/SidewalkWebpage/wiki/Style-Guide) as well.

MohitSoni11 replied:
Done!

misaugstad commented on app/views/navbar.scala.html:
```diff
@@ -11,9 +11,30 @@
+            <div class = "test-server-banner">
+                <div class = "test-server-banner-content">
+                    <div class = "test-server-banner-text">
+                        <strong>WARNING!</strong> You are on the Test Server!
```
Can you change this from "the Test Server" to "a test server"

MohitSoni11 replied:
Done!

misaugstad reviewed (approved):
I love this, thanks @MohitSoni11! I added a couple small comments throughout the code, but the main thing I wanted to bring up is that we won't actually see this warning on mobile because we don't show the navbar there. The only place where we really need it is at /mobile (though /signInMobile and /signUpMobile would be fine as well). So you may have to relocate this code and do a bit of reworking to get it to work everywhere.

misaugstad commented:
Also I accidentally clicked 'approve' instead of 'request changes', my bad!

MohitSoni11 commented:
@misaugstad In the last commit, I have changed the style of the banner to the picture below because personally I thought this was cleaner in terms of looks (notice I removed the button). Please inform me if you would like to change it back, or to another style.

![image](https://user-images.githubusercontent.com/92624795/184458316-cec6986a-301a-4587-927f-0b7345569694.png)

Additionally, I have cut the html from the navbar.html file and added it into the signUpMobile.html file and signInMobile.html with the css in mobileAuthenticate.css instead of main.css. This was done so that the banner would be visible when signing in or signing up in mobile.
- I have tested this code using Chrome Developer Tools because (as you know), I am not able to access the mobile version of the site

misaugstad commented:
Oh, when I said
>The only place where we really need it is at /mobile (though /signInMobile and /signUpMobile would be fine as well)

I was specifically referring to mobile since you didn't have it there before. I didn't mean that those are the _only_ pages where we should have the test banner.

Before, you had it on every page on the desktop site, but none on the mobile site. We really just want it on every page!

So when I said 
>So you may have to relocate this code and do a bit of reworking to get it to work everywhere.

I meant that you could take the code out of `navbar.scala.html`, but you would then _import_ in there, since we still want the banner everywhere! :grin:

misaugstad commented:
Oh, and I would love to still have a button to close it! I want for the test servers to be able to look identical to prod when I'm testing out new code. Maybe when closing the banner can you also save that in a cookie? How hard would that be to do?

jonfroehlich commented:
Maybe update the warning to: "WARNING: You are on the test server. Your data will not be used."

jonfroehlich commented:
Looks awesome. Thank you!
